src/main/clojure/clara/rules/engine/helpers.cljc

```diff
@@ -0,0 +1,66 @@
+(ns clara.rules.engine.helpers
+  (:require
+    [clara.rules.engine.impl :as impl]
+    [clara.rules.engine.wme :as wme]))
+
+(defn has-keys?
+  "Returns true if the given map has all of the given keys."
+  [m keys]
+  (every? (partial contains? m) keys))
+
+(defn matches-some-facts?
+  "Returns true if the given token matches one or more of the given elements."
+  [token elements join-filter-fn condition]
+  (some (fn [{:keys [fact]}]
+          (join-filter-fn token fact (:env condition)))
+        elements))
+
+(defn send-accumulated
```

Comment:
Understood. The main ones I think of as implementation details are send-accumulated and retract-accumulated, which are pretty specific to that namespace. I don't mind exposing things like has-keys here.